server/src/test/java/com/paypal/selion/grid/servlets/BaseGridRegistyServletTest.java

```diff
@@ -0,0 +1,132 @@
+/*-------------------------------------------------------------------------------------------------------------------*\
+|  Copyright (C) 2016 PayPal                                                                                          |
+|                                                                                                                     |
+|  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance     |
+|  with the License.                                                                                                  |
+|                                                                                                                     |
+|  You may obtain a copy of the License at                                                                            |
+|                                                                                                                     |
+|       http://www.apache.org/licenses/LICENSE-2.0                                                                    |
+|                                                                                                                     |
+|  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed   |
+|  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for  |
+|  the specific language governing permissions and limitations under the License.                                     |
+\*-------------------------------------------------------------------------------------------------------------------*/
+
+package com.paypal.selion.grid.servlets;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.openqa.grid.common.RegistrationRequest;
+import org.openqa.grid.internal.Registry;
+import org.openqa.grid.internal.RemoteProxy;
+import org.openqa.grid.internal.utils.CapabilityMatcher;
+import org.openqa.grid.internal.utils.DefaultCapabilityMatcher;
+import org.openqa.grid.internal.utils.GridHubConfiguration;
+import org.openqa.grid.selenium.proxy.DefaultRemoteProxy;
+import org.openqa.grid.web.Hub;
+import org.openqa.selenium.net.NetworkUtils;
+import org.openqa.selenium.net.PortProber;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+import com.google.common.base.Preconditions;
+
+public class BaseGridRegistyServletTest {
+    protected static String ipAddress;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Possible unsafe assignment to a non-final static field in a constructor.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2102631886/issues/source?bid=2928425&fileBranchId=3141724#l41)